### PR TITLE
use native ios message handler for attachments, better speed

### PR DESCRIPTION
**Pull request by brettharrisonzya**

### CODE REVIEW

File: src/Couchbase.Lite.Shared/Replication/AttachmentPuller.cs [hunk 1]
```diff
@@ -193,6 +180,21 @@ internal void PullRemoteAttachment (AttachmentRequest req)
                         // are still revisions waiting to be pulled:
                         --_httpConnectionCount;
                         PullRemoteAttachments ();
+
+                        if (e == null)
+                        {
+                            var bytes = req.GetBuffer();
+                            if (bytes == null)
+                            {
+                                Log.E (TAG, "Error pulling remote attachment GettBuffer == null");
```
Comment (zippo227) [inline src/Couchbase.Lite.Shared/Replication/AttachmentPuller.cs]: GetBuffer

File: src/Couchbase.Lite.Shared/Replication/RemoteSession.cs [hunk 2]
```diff
@@ -477,16 +513,24 @@ internal void SendAsyncMultipartRequest(HttpMethod method, string relativePath,
             var message = new HttpRequestMessage (method, url);
             message.Headers.Add ("Accept", "*/*");
 
+#if __IOS__  // hack to get a client with NSUrlSessionHandler which is much faster
+            var client = _iosHttpClient;
```
Comment (zippo227) [inline src/Couchbase.Lite.Shared/Replication/RemoteSession.cs:517]: Would it do this automatically if we switched our project's iOS build settings for HttpClient from Managed(default) to NSUrlSession? Then we wouldn't need this part.

**Review by zippo227 — COMMENTED** [review]
Couple of comments.

File: src/Couchbase.Lite.Shared/Replication/RemoteSession.cs (hunk 2 above)
Comment (brettharrisonzya) [reply src/Couchbase.Lite.Shared/Replication/RemoteSession.cs:517]: That setting does not change how dlls work.  Sadly.

Comment (zippo227) [reply src/Couchbase.Lite.Shared/Replication/RemoteSession.cs:517]: That stinks. I'll keep my eyes peeled for some other way. That's a big problem.

Comment (brettharrisonzya) [reply src/Couchbase.Lite.Shared/Replication/RemoteSession.cs:517]: some libs let you pass in a httpClient (FFImageLoader does)

**Pull request merged**